File: kurs programowania/lista3/zadanie1/Prostokat.java
public class Prostokat extends Czworokat {

    public Prostokat(int a, int b) {
        super(a, a, b, b, 90);
    }

    public double pole() {

        return bok1 * bok3;
    }

    public double obwod() {

        return bok1 + bok2 + bok3 + bok4;
    }
}

File: kurs programowania/lista3/cpp/Prostokat.cpp
#include "Prostokat.hpp"

#include <iostream>

Prostokat::Prostokat(int a, int b) : Czworokat(a, b, a, b, 90) {}

double Prostokat::pole()
{
    return a * b;
}

double Prostokat::obwod()
{
    return 2 * a + 2 * b;
}
